context/index.js

```diff
@@ -0,0 +1,89 @@
+let contexts = new Map()
+
+let _createContext = name => {
+  let context = {
+    // store [c]opies
+    c: new Map(),
+    n: name,
+    // listener [q]ueue
+    q: [],
+    // store [s]tates
+    s: new Map()
+  }
+  contexts.set(name, context)
+  return context
+}
+export let globalContext = _createContext()
+export let createContext = name => {
+  // global context is [d]ead
+  globalContext.d = true
+  return _createContext(name)
+}
+export let resetContext = name => {
+  if (name) {
+    contexts.delete(name)
+  } else {
+    contexts.clear()
+    globalContext.d = false
+    delete globalContext.ta
+  }
+}
+export let getContext = name => contexts.get(name)
+
+// lazily initialize store state in context
+export function getStoreState(thisStore, originalStore) {
+  if (thisStore.ctx.d) {
+    if (process.env.NODE_ENV !== 'production') {
+      throw new Error(
+        `You can't mix global context and custom contexts, that is probably an error. ` +
+          'Please, pass the correct context into `withContext`.'
+      )
+    }
+    // eslint-disable-next-line no-throw-literal
```

Comment:
We need a comment here about how we are planning to use it.

Also, just disable `no-throw-literal` globally if we are going to use this signal method.

Reply:
Tbh, I suspect it shouldn't fire at all. We have an elaborate error message for development, but it seemed to me that it shouldn't change the behavior in production. But I wanted the throw to affect bundle size as little as possible. I think, throwing a number is the smallest in terms of chars. I don't expect anyone to catch it or work with it in any way.

Reply:
In this case I think it is better to throw `new Error('no global context')` or `new Error('CODE')`